src/server/meme.js

```diff
@@ -142,4 +143,56 @@ module.exports = psql => ({
       };
     }
   },
+
+  getMemes: async (username, earliestId, user) => {
+    try {
+      let memes;
+
+      /*
+        Get all memes if username not provided. Pagination of memes identical to
+        meme comments above.
+      */
+      if (!username) {
+        if (earliestId) {
+          memes = await baseMemeQuery(psql, user)
+            .where('memes.meme_id', '<', earliestId)
+            .orderBy('meme_id', 'desc')
+            .limit(MEME_PAGE_SIZE);
+        } else {
+          memes = await baseMemeQuery(psql, user)
+            .orderBy('meme_id', 'desc')
+            .limit(MEME_PAGE_SIZE);
+        }
+        return {
+          isSuccessful: true,
+          value: memes,
+        };
+      }
+      // else get memes belonging to the provided user
+      if (earliestId) {
+        memes = await baseMemeQuery(psql, user)
+          .innerJoin('users as users2', 'memes.user_id', 'users2.user_id')
+          .where('users2.username', username)
```

Comment:
The base query already selects `users.username`, would that field work in this where clause? I think it should, but I may be wrong. That field does indicate the author of the meme. Then we could avoid the extra join.

Also, I don't know but it might help for this to be an `.andWhere()` but I don't know if that's right.

Reply:
I can try again with those suggestions, but I did test everything and everything seems to return as expected.

Reply:
@nxtman123 you're right. Removed inner join. Also changed to .andWhere for all queries, everything works.